### PR TITLE
optimize: refactor loadBalance config from registry.conf to file.conf.

### PR DESCRIPTION
**Pull request by ls9527**
### Ⅰ. Describe what this PR did
loadbalance的配置在 registry.conf 文件， 这个文件属于配置中心的基础配置， 除了指定方式和基础信息以外，其他信息不应该在这个文件。 把loadBalance改到核心配置文件中(file.conf).  

例如随机负载均衡
loadBalance = {
    type = "RandomLoadBalance"
}

还有一致性hash负载均衡
loadBalance = {
    type = "ConsistentHashLoadBalance"
    visualNodes=10
}

### CODE REVIEW

**Comment by caohdgege** [comment]
[ERROR] /home/travis/build/seata/seata/seata-spring-boot-starter/src/test/java/io/seata/spring/boot/autoconfigure/PropertiesTest.java:[223,59] cannot find symbol
  symbol:   class LoadBalanceProperties
  location: class io.seata.spring.boot.autoconfigure.PropertiesTest
[ERROR] /home/travis/build/seata/seata/seata-spring-boot-starter/src/test/java/io/seata/spring/boot/autoconfigure/PropertiesTest.java:[224,42] cannot find symbol
  symbol:   class LoadBalanceProperties
  location: class io.seata.spring.boot.autoconfigure.PropertiesTest

**Comment by caohdgege** [comment]
处理一下ci的报错

File: seata-spring-boot-starter/src/test/java/io/seata/spring/boot/autoconfigure/PropertiesTest.java [hunk 1]
```diff
@@ -36,15 +36,7 @@
...
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryConsulProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryCustomProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryEtcd3Properties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryEurekaProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryNacosProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryRedisProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistrySofaProperties;
-import io.seata.spring.boot.autoconfigure.properties.registry.RegistryZooKeeperProperties;
+import io.seata.spring.boot.autoconfigure.properties.registry.*;
```
Comment (a364176773) [inline seata-spring-boot-starter/src/test/java/io/seata/spring/boot/autoconfigure/PropertiesTest.java:39]: * is not allowed

File: server/src/main/resources/registry.conf [hunk 2]
```diff
@@ -1,8 +1,7 @@
 registry {
   # file 、nacos 、eureka、redis、zk、consul、etcd3、sofa
   type = "file"
-  loadBalance = "RandomLoadBalance"
-  loadBalanceVirtualNodes = 10
+
```
Comment (a364176773) [inline server/src/main/resources/registry.conf:4]: Blank lines can be removed

**Review by a364176773 — CHANGES_REQUESTED** [review]
In addition, please in server/file. Conf. Example, script/server/conf and client/conf&spring, add configuration information

**Comment by a364176773** [comment]
@ph3636 PTAL

**Comment by ph3636** [comment]
Search it according to the keyword (loadBalanceVirtualNodes), or check it through git records, there are still many places that have not been changed.
根据关键词（loadBalanceVirtualNodes）搜一下，或者通过git记录看下，还有好多地方没改。

**Comment by ls9527** [comment]
@ph3636  好的，  修改了。 
@caohdgege 已经提交了遗漏提交的文件

**Comment by ph3636** [comment]
I think this configuration is more appropriate to put in the client, the server will not use this configuration
我觉得这个配置放到client中比较合适，server不会用到这个配置

**Comment by ls9527** [comment]
> I think this configuration is more appropriate to put in the client, the server will not use this configuration
> 我觉得这个配置放到client中比较合适，server不会用到这个配置

有道理阿， 只有TM，RM才需要负载均衡。  TC都知道发给谁了，就不用均衡了。   那我在改改

**Comment by ph3636** [comment]
Change these files
改一下这些文件
script/client/spring/application.properties
script/client/spring/application.yml
META-INF/additional-spring-configuration-metadata.json

No configuration in these files
下面这些文件中可以不加配置
script/server/config

File: discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java [hunk 3]
```diff
@@ -36,8 +35,14 @@
 @LoadLevel(name = "ConsistentHashLoadBalance")
 public class ConsistentHashLoadBalance extends AbstractLoadBalance {
 
-    private static final String VIRTUAL_NODES = FILE_ROOT_REGISTRY + FILE_CONFIG_SPLIT_CHAR + "loadBalanceVirtualNodes";
-    private static final int VIRTUAL_NODES_NUM = ConfigurationFactory.CURRENT_FILE_INSTANCE.getInt(VIRTUAL_NODES, VIRTUAL_NODES_DEFAULT);
+    /**
+     * The constant LOAD_BALANCE_TYPE.
```
Comment (ph3636) [inline discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java:39]: Change this name
改一下这个名字

File: script/server/config/file.properties [hunk 4]
```diff
@@ -25,3 +25,5 @@ store.redis.database=0
 store.redis.minConn=1
 store.redis.maxConn=10
 store.redis.queryLimit=100
+loadBalance.type=RandomLoadBalance
+loadBalance.consistentHash.virtualNodes=10
```
Comment (ph3636) [inline script/server/config/file.properties:29]: These are redundant
这些是多余

File: script/server/config/file.yml [hunk 5]
```diff
@@ -44,4 +44,9 @@ store:
     minConn: 1
     maxConn: 10
     queryLimit: 100
+loadBalance:
+  type: RandomLoadBalance
+  consistentHash:
+    virtualNodes: 10
```
Comment (ph3636) [inline script/server/config/file.yml:50]: These are redundant
这些是多余

Comment (ls9527) [reply script/server/config/file.yml]: 多谢大神， 已将多余的部分删除。  将loadBalance的 type 和  visualNodes 区分开了。

File: discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java (hunk 3 above)
Comment (ph3636) [reply discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java:39]: Change the name or delete it

**Review by ph3636 — APPROVED** [review]
LGTM

File: discovery/seata-discovery-core/src/test/resources/file.conf [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#
```
Comment (a364176773) [inline discovery/seata-discovery-core/src/test/resources/file.conf:1]: unwanted file

File: script/server/config/file.conf [hunk 7]
```diff
@@ -49,5 +49,4 @@ store {
     maxConn = 10
     queryLimit = 100
   }
-
-}
+}
```
Comment (a364176773) [inline script/server/config/file.conf:52]: unnecessary change

File: script/server/config/file.properties [hunk 8]
```diff
@@ -24,4 +24,4 @@ store.redis.password=
 store.redis.database=0
 store.redis.minConn=1
 store.redis.maxConn=10
-store.redis.queryLimit=100
+store.redis.queryLimit=100
```
Comment (a364176773) [inline script/server/config/file.properties:27]: unnecessary change

File: script/server/config/file.yml [hunk 9]
```diff
@@ -43,5 +43,4 @@ store:
     database: 0
     minConn: 1
     maxConn: 10
-    queryLimit: 100
-
+    queryLimit: 100
```
Comment (a364176773) [inline script/server/config/file.yml:46]: unnecessary change

File: server/src/main/resources/file.conf [hunk 10]
```diff
@@ -50,4 +50,4 @@ store {
     queryLimit = 100
   }
 
-}
+}
```
Comment (a364176773) [inline server/src/main/resources/file.conf:53]: unnecessary change

**Review by a364176773 — CHANGES_REQUESTED** [review]

File: discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java (hunk 3 above)
Comment (ls9527) [reply discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java]: 一直以为改了，今天核对才发现没改。  刚刚改掉了

File: script/server/config/file.properties (hunk 4 above)
Comment (ls9527) [reply script/server/config/file.properties]: 多余的删掉了

**Comment by a364176773** [comment]
请解决代码冲突

**Review by a364176773 — APPROVED** [review]
LGTM

File: changes/en-us/1.5.0.md [hunk 11]
```diff
@@ -59,7 +59,7 @@
   - [[#3420](https://github.com/seata/seata/pull/3420)] optimize enumerated classes and add unit tests	
   - [[#3436](https://github.com/seata/seata/pull/3436)] optimize typo in SQLType class 	
   - [[#3439](https://github.com/seata/seata/pull/3439)] adjust the order of springApplicationContextProvider
-
+  - [[#3248](https://github.com/seata/seata/pull/3248)] optimize the config of load-balance migration to belong the client node
```
Comment (l81893521) [inline changes/en-us/1.5.0.md:62]: migrate instead of migration, migration is not a verb.

**Review by l81893521 — APPROVED** [review]
LGTM

**Review by xingfudeshi — APPROVED** [review]
LGTM.

File: discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java [hunk 12]
```diff
@@ -37,8 +35,14 @@
 @LoadLevel(name = CONSISTENT_HASH_LOAD_BALANCE)
 public class ConsistentHashLoadBalance extends AbstractLoadBalance {
 
-    private static final String VIRTUAL_NODES = FILE_ROOT_REGISTRY + FILE_CONFIG_SPLIT_CHAR + "loadBalanceVirtualNodes";
-    private static final int VIRTUAL_NODES_NUM = ConfigurationFactory.CURRENT_FILE_INSTANCE.getInt(VIRTUAL_NODES, VIRTUAL_NODES_DEFAULT);
+    /**
+     * The constant LOAD_BALANCE_CONSISTENT_HASH_VISUAL_NODES.
+     */
+    public static final String LOAD_BALANCE_CONSISTENT_HASH_VISUAL_NODES = LOAD_BALANCE_PREFIX + "visualNodes";
```
Comment (wangliang181230) [inline discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/ConsistentHashLoadBalance.java:41]: @xingfudeshi 
配置名 `visualNodes` 与 `LoadBalanceProperties` 类里的field `virtualNodes` 名不一致，导致在 `SpringBootConfigurationProvider` 下，获取字段时不存在，导致返回为null。
也就是说在springboot下，配置了`seata.client.load-balance.visual-nodes` 或 `seata.client.load-balance.virtual-nodes`都会导致配置读取不到。

**Pull request merged**